src/CborParseNode.cs

```diff
@@ -403,32 +428,58 @@ public IEnumerable<T> GetCollectionOfPrimitiveValues<T>()
             string str => str,
             int[] intArray => intArray,
             string[] strArray => strArray,
-            object?[] arrayValue => arrayValue.Select(TryGetAnything).ToArray(),
-            List<object?> listValue => listValue.Select(TryGetAnything).ToArray(),
+            object?[] arrayValue => ProcessArray(arrayValue),
+            List<object?> listValue => ProcessList(listValue),
             Dictionary<string, object?> dictionaryValue => dictionaryValue,
             CborParseNode node => TryGetAnything(node.value),
             bool b => b,
             null => null,
             _ => throw new InvalidOperationException($"unexpected additional value type during deserialization json kind : {val.GetType()}"),
         };
 
+        private static object?[] ProcessArray(object?[] arrayValue)
+        {
+            var newArray = new object?[arrayValue.Length];
+            for(int index = 0; index < arrayValue.Length; index++)
+            {
+                newArray[index] = TryGetAnything(arrayValue[index]);
+            }
+            return newArray;
+        }
+
+        private static object?[] ProcessList(List<object?> listValue)
+        {
+            var newArray = new object?[listValue.Count];
+            for(int index = 0; index < listValue.Count; index++)
+            {
+                newArray[index] = TryGetAnything(listValue[index]);
+            }
+            return newArray;
+        }
+
         /// <summary>
         /// Get the child node of the specified identifier
         /// </summary>
         /// <param name="identifier">The identifier of the child node</param>
         /// <returns>An instance of <see cref="IParseNode"/></returns>
-        public IParseNode? GetChildNode(string identifier)
```

Comment:
Looks like `GetChildNode` got dropped which is a contract defined by the parent interace.